src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -49,8 +49,23 @@ const std::string kDescriptorMetadataFile =
     "GPBMetadata/Google/Protobuf/Internal/Descriptor.php";
 const std::string kDescriptorDirName = "Google/Protobuf/Internal";
 const std::string kDescriptorPackageName = "Google\\Protobuf\\Internal";
-const char* const kReservedNames[] = {"ARRAY", "Empty", "ECHO"};
-const int kReservedNamesSize = 3;
+const char* const kReservedNames[] = {
```

Comment:
There is a slightly different behavior between which classes are permitted as class names and which are permitted as constants. The names `int`, `float`, `bool`, `string`, `true`, `false`, `null`, `void` and `iterable` are valid as PHP constants, but not as class names.

We could support allowing these by supporting a reduced set of reserved names that would be checked here instead of the full set checked in ClassNamePrefix: https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/php/php_generator.cc#L984